src/components/GeneralInfo/DataCollectorsCard/DataCollectorsCard.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable camelcase */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import LoadingCard from '../LoadingCard';
+import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
+import { TableComposable, Thead, Tr, Th, Tbody, Td, TableVariant, ExpandableRowContent } from '@patternfly/react-table';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { registered } from '../../../Utilities/index';
+
+const getDefaultCollectors = (entity) =>
+    registered?.filter(reporter => reporter.label !== 'insights-client not connected')
+    .map(reporter => ({
+        name: reporter.label,
+        status: entity?.per_reporter_staleness[reporter.value]?.check_in_succeeded,
+        updated: entity?.per_reporter_staleness[reporter.value]?.last_check_in,
+        details: {
+            name: reporter.idName,
+            id: entity?.[reporter.idValue]
+        }
+    }));
+
+const DataCollectorsCard = ({
+    detailLoaded,
+    collectors,
+    entity
+}) => {
+    const [expandedNames, setExpandedNames] = useState([]);
+    const setExpanded = (collector, isExpanding = true) =>
+        setExpandedNames(prevExpanded => {
+            const otherExpandedNames = prevExpanded.filter(r => r !== collector.name);
+            return isExpanding ? [...otherExpandedNames, collector.name] : otherExpandedNames;
+        });
+    const isExpanded = (collector) => expandedNames.includes(collector.name);
+    const data = collectors ?? getDefaultCollectors(entity);
```

Comment:
I like this approach! If someone wants to change the data they can pass in custom collectors. Could be good to document this somewhere :thinking: